From 81e52f498651cadf4bc3c8c69d189af16e947ca4 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Fri, 28 Sep 2018 17:19:05 +0900
Subject: [PATCH 102/909] clk: renesas: r8a77{95,965,990}: Remove INTC-AP and
 INTC-EX clocks

commit 2826ead86b471df62b9f3214ffb43b70e0759152 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

According to the R-Car Gen3 Hardware Manual Rev 1.00, the the module clock
status, module clock control, and reset control bits of the INTC-AP and
INTC-EX modules on R-Car H3-W, R-Car M3-N and R-Car E3 are removed.

Continue to change the clock properties of the gic and irqc nodes in the
device tree.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/renesas/r8a7795-cpg-mssr.c  | 12 ------------
 drivers/clk/renesas/r8a77965-cpg-mssr.c | 10 ----------
 drivers/clk/renesas/r8a77990-cpg-mssr.c | 10 ----------
 3 files changed, 32 deletions(-)

diff --git a/drivers/clk/renesas/r8a7795-cpg-mssr.c b/drivers/clk/renesas/r8a7795-cpg-mssr.c
index 4e63c02474cf..1718fec9f9f2 100644
--- a/drivers/clk/renesas/r8a7795-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a7795-cpg-mssr.c
@@ -151,8 +151,6 @@ static struct mssr_mod_clk r8a7795_mod_clks[] __initdata = {
 	DEF_MOD("usb-dmac0",		 330,	R8A7795_CLK_S3D1),
 	DEF_MOD("usb-dmac1",		 331,	R8A7795_CLK_S3D1),
 	DEF_MOD("rwdt",			 402,	R8A7795_CLK_R),
-	DEF_MOD("intc-ex",		 407,	R8A7795_CLK_CP),
-	DEF_MOD("intc-ap",		 408,	R8A7795_CLK_S0D3),
 	DEF_MOD("audmac1",		 501,	R8A7795_CLK_S1D2),
 	DEF_MOD("audmac0",		 502,	R8A7795_CLK_S1D2),
 	DEF_MOD("drif7",		 508,	R8A7795_CLK_S3D2),
@@ -272,11 +270,6 @@ static struct mssr_mod_clk r8a7795_mod_clks[] __initdata = {
 	DEF_MOD("scu-src0",		1031,	MOD_CLK_ID(1017)),
 };
 
-static const unsigned int r8a7795_crit_mod_clks[] __initconst = {
-	MOD_CLK_ID(408),	/* INTC-AP (GIC) */
-};
-
-
 /*
  * CPG Clock Data
  */
@@ -351,7 +344,6 @@ static const struct mssr_mod_reparent r8a7795es1_mod_reparent[] __initconst = {
 	{ MOD_CLK_ID(217), R8A7795_CLK_S3D1 },	/* SYS-DMAC2 */
 	{ MOD_CLK_ID(218), R8A7795_CLK_S3D1 },	/* SYS-DMAC1 */
 	{ MOD_CLK_ID(219), R8A7795_CLK_S3D1 },	/* SYS-DMAC0 */
-	{ MOD_CLK_ID(408), R8A7795_CLK_S3D1 },	/* INTC-AP */
 	{ MOD_CLK_ID(501), R8A7795_CLK_S3D1 },	/* AUDMAC1 */
 	{ MOD_CLK_ID(502), R8A7795_CLK_S3D1 },	/* AUDMAC0 */
 	{ MOD_CLK_ID(523), R8A7795_CLK_S3D4 },	/* PWM */
@@ -467,10 +459,6 @@ const struct cpg_mssr_info r8a7795_cpg_mssr_info __initconst = {
 	.num_mod_clks = ARRAY_SIZE(r8a7795_mod_clks),
 	.num_hw_mod_clks = 12 * 32,
 
-	/* Critical Module Clocks */
-	.crit_mod_clks = r8a7795_crit_mod_clks,
-	.num_crit_mod_clks = ARRAY_SIZE(r8a7795_crit_mod_clks),
-
 	/* Callbacks */
 	.init = r8a7795_cpg_mssr_init,
 	.cpg_clk_register = rcar_gen3_cpg_clk_register,
diff --git a/drivers/clk/renesas/r8a77965-cpg-mssr.c b/drivers/clk/renesas/r8a77965-cpg-mssr.c
index 3cbdd8e873d7..0e9c0e754532 100644
--- a/drivers/clk/renesas/r8a77965-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a77965-cpg-mssr.c
@@ -142,8 +142,6 @@ static const struct mssr_mod_clk r8a77965_mod_clks[] __initconst = {
 	DEF_MOD("usb-dmac1",		331,	R8A77965_CLK_S3D1),
 
 	DEF_MOD("rwdt",			402,	R8A77965_CLK_R),
-	DEF_MOD("intc-ex",		407,	R8A77965_CLK_CP),
-	DEF_MOD("intc-ap",		408,	R8A77965_CLK_S0D3),
 
 	DEF_MOD("audmac1",		501,	R8A77965_CLK_S1D2),
 	DEF_MOD("audmac0",		502,	R8A77965_CLK_S1D2),
@@ -246,10 +244,6 @@ static const struct mssr_mod_clk r8a77965_mod_clks[] __initconst = {
 	DEF_MOD("scu-src0",		1031,	MOD_CLK_ID(1017)),
 };
 
-static const unsigned int r8a77965_crit_mod_clks[] __initconst = {
-	MOD_CLK_ID(408),	/* INTC-AP (GIC) */
-};
-
 /*
  * CPG Clock Data
  */
@@ -331,10 +325,6 @@ const struct cpg_mssr_info r8a77965_cpg_mssr_info __initconst = {
 	.num_mod_clks		= ARRAY_SIZE(r8a77965_mod_clks),
 	.num_hw_mod_clks	= 12 * 32,
 
-	/* Critical Module Clocks */
-	.crit_mod_clks		= r8a77965_crit_mod_clks,
-	.num_crit_mod_clks	= ARRAY_SIZE(r8a77965_crit_mod_clks),
-
 	/* Callbacks */
 	.init			= r8a77965_cpg_mssr_init,
 	.cpg_clk_register	= rcar_gen3_cpg_clk_register,
diff --git a/drivers/clk/renesas/r8a77990-cpg-mssr.c b/drivers/clk/renesas/r8a77990-cpg-mssr.c
index 8802ad451db0..9a517c2c8ee6 100644
--- a/drivers/clk/renesas/r8a77990-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a77990-cpg-mssr.c
@@ -149,8 +149,6 @@ static const struct mssr_mod_clk r8a77990_mod_clks[] __initconst = {
 	DEF_MOD("usb-dmac1",		 331,	R8A77990_CLK_S3D1),
 
 	DEF_MOD("rwdt",			 402,	R8A77990_CLK_R),
-	DEF_MOD("intc-ex",		 407,	R8A77990_CLK_CP),
-	DEF_MOD("intc-ap",		 408,	R8A77990_CLK_S0D3),
 
 	DEF_MOD("audmac0",		 502,	R8A77990_CLK_S1D2),
 	DEF_MOD("drif7",		 508,	R8A77990_CLK_S3D2),
@@ -239,10 +237,6 @@ static const struct mssr_mod_clk r8a77990_mod_clks[] __initconst = {
 	DEF_MOD("scu-src0",		1031,	MOD_CLK_ID(1017)),
 };
 
-static const unsigned int r8a77990_crit_mod_clks[] __initconst = {
-	MOD_CLK_ID(408),	/* INTC-AP (GIC) */
-};
-
 /*
  * CPG Clock Data
  */
@@ -288,10 +282,6 @@ const struct cpg_mssr_info r8a77990_cpg_mssr_info __initconst = {
 	.num_mod_clks = ARRAY_SIZE(r8a77990_mod_clks),
 	.num_hw_mod_clks = 12 * 32,
 
-	/* Critical Module Clocks */
-	.crit_mod_clks = r8a77990_crit_mod_clks,
-	.num_crit_mod_clks = ARRAY_SIZE(r8a77990_crit_mod_clks),
-
 	/* Callbacks */
 	.init = r8a77990_cpg_mssr_init,
 	.cpg_clk_register = rcar_gen3_cpg_clk_register,
-- 
2.17.1

